spectrum-network/src/protocol_handler.rs

```diff
@@ -138,7 +139,7 @@ where
     TBehaviour: ProtocolBehaviour + Unpin,
     TNetwork: NetworkAPI + Unpin,
 {
-    type Item = ();
+    type Item = <TBehaviour::TProto as ProtocolSpec>::TMessage;
```

Comment:
Make the `Stream` impl for `ProtocolHandler` return the protocol messages, so we can capture them in our tests.

Reply:
Shouldn't we make that feature optional for tests? As there is no need to pass those messages around in the prod environment.

Reply:
Done.